src/config/TokenContext.tsx

```diff
@@ -0,0 +1,148 @@
+  loadingProcessInfo: '',
+  navigateToLogin: null,
+};
+
+const TokenContext = createContext<TokenContextValues>(defaultValue);
+
+export const TokenContextProvider = ({ children }: ContextProviderProps) => {
+  const {
+    accessToken: defaultAccessToken,
+    loading: defaultLoading,
+    loadingProcessInfo: defaultLoadingProcessInfo,
+    navigateToLogin: defaultNavigateToLogin,
+  } = defaultValue;
+
+  const [accessToken, setAccessToken] = useState(defaultAccessToken);
+  const [timeLeft, setTimeLeft] = useState(3600);
+  const [refreshToken, setRefreshToken] = useState('');
+  const [navigateToLogin, setNavigatorToLogin] = useState(defaultNavigateToLogin);
+  const [loading, setLoading] = useState(defaultLoading);
+  const [loadingProcessInfo, setLoadingProcessInfo] = useState(defaultLoadingProcessInfo);
+
+  //* Fetch new access token
+  const getAccessToken = async (token: string) => {
+    setLoading(true);
+    setLoadingProcessInfo('Fetching Token');
+    const {
+      accountData: { grantType },
+      endpoints: {
+        account: { requestAccessToken },
+      },
+    } = ApiConstants;
+
+    const url = requestAccessToken;
+    const body: RequestAccessTokenBody = {
+      grant_type: grantType.token,
+      refresh_token: token,
+    };
+
+    const response = await _postAccount<AuthAccessTokenResponse, RequestAccessTokenBody>(url, body);
+
+    if (!response.success) {
+      setLoading(false);
+      const { message } = response.error;
+      Toast.show({ text1: message });
+      return;
+    }
+
+    const { access_token, expires_in, token_type } = response.responseData;
+    saveAccessToken(`${token_type} ${access_token}`, expires_in);
+
+    setLoading(false);
+  };
+
+  //* Set token expiration in timer
+  const updateTokenExpirationTimer = (expiresAt: string) => {
+    const expirationTime = new Date(expiresAt).getTime();
+    const remainingTime = Math.max(0, Math.floor((expirationTime - Date.now()) / 1000));
+    setTimeLeft(remainingTime);
+  };
+
+  //* Load Stored Data from Storage
+  const loadStore = async () => {
+    setLoading(true);
+    setLoadingProcessInfo('Fetching Data');
+    const {
+      accessToken: accessTokenKey,
+      refreshToken: refreshTokenKey,
+      tokenExpiresIn: tokenExpiresInKey,
+    } = StorageKey;
+
+    const storedRefreshToken = (await StorageManager.getStoreValue<string>(refreshTokenKey)) ?? '';
+    setRefreshToken(storedRefreshToken);
+
+    setNavigatorToLogin(!storedRefreshToken);
+    if (!storedRefreshToken) return;
+
+    const storedAccessToken = (await StorageManager.getStoreValue<string>(accessTokenKey)) ?? '';
+    setAccessToken(storedAccessToken);
+
+    const storedExpiresIn = (await StorageManager.getStoreValue<string>(tokenExpiresInKey)) ?? '';
+    updateTokenExpirationTimer(storedExpiresIn);
+
+    setLoading(false);
+  };
+
+  //* Use Effects
+  useEffect(() => {
+    loadStore();
+  }, []);
+
+  useEffect(() => {
+    if (refreshToken && timeLeft <= 0) {
```

Comment:
explain `timeLeft`, and mention why you are handling this manually.

Reply:
i am getting expiry of access token that i am setting in time left

My logic works like this:
If user is still on the app and it token has been expired,
then it will automatically generate a new access toke and the timer will again reset to `expires_in` from response.

this way it will be better for user also
and better for developer as he did not need to handle the 401 code in every api call